lib/Sema/CSSimplify.cpp

```diff
@@ -9843,6 +9843,12 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyFixConstraint(
     auto impact = 1;
 
     auto locator = fix->getLocator();
+
+    // If the locator points to a contextual type let's increase the impact
+    // slightly trying to avoid ambiguity.
+    if (locator->isForContextualType())
+      impact = 2;
```

Comment:
I wanted to treat "contextual mismatch" as just a generic fix which means that there is no way to reconcile the types of expression with context, so I didn't really want to make it more impactful then a regular requirement failure.

Reply:
Humm, I think you right ... initially, I thought that except for the key path one, the all improved a bit. But seeing your comments I agree. The intention here was just to disambiguate in the case of the SR by favoring the closure contextual failures instead of the contextual type.